### PR TITLE
Adding pokemons on mainRV MVP

### PR DESCRIPTION
**Pull request by Fredest9112**
## **Proposed changes**

- Added initial functionality, showing pokemons from db to MainRV

## **Checklist**

- Added methods on model and presenter
- Added adapter for MainRV

### CODE REVIEW

File: app/src/main/java/com/learningkotlin/pokedex/interfaces/IPokemonModel.kt [hunk 1]
```diff
@@ -1,4 +1,11 @@
 package com.learningkotlin.pokedex.interfaces
 
+import androidx.lifecycle.LiveData
+import com.learningkotlin.pokedex.repository.database.Pokemon
+
 interface IPokemonModel {
+    suspend fun savePokemonInfo()
+    fun loadPokemon(): LiveData<List<Pokemon>>
+//    fun loadPokemonByType(type:String): LiveData<List<Pokemon>>
+//    fun loadPokemonByQuery(query:String): LiveData<List<Pokemon>>
```
Comment (fredestReviewer) [inline app/src/main/java/com/learningkotlin/pokedex/interfaces/IPokemonModel.kt:10]: Remove commented Code

File: app/src/main/java/com/learningkotlin/pokedex/model/PokemonModel.kt [hunk 2]
```diff
@@ -1,4 +1,87 @@
...
+
+    override fun loadPokemon(): LiveData<List<Pokemon>> = liveData {
+        repository.pokemon.collect {
+            emit(it)
+        }
+    }
+
+//    override fun loadPokemonByType(type: String): LiveData<List<Pokemon>> {
+//        TODO("Not yet implemented")
+//    }
```
Comment (fredestReviewer) [inline app/src/main/java/com/learningkotlin/pokedex/model/PokemonModel.kt:91]: Remove Commented Code

**Review by fredestReviewer — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/learningkotlin/pokedex/interfaces/IPokemonModel.kt (hunk 1 above)
Comment (Fredest9112) [reply app/src/main/java/com/learningkotlin/pokedex/interfaces/IPokemonModel.kt]: commented code removed

File: app/src/main/java/com/learningkotlin/pokedex/model/PokemonModel.kt (hunk 2 above)
Comment (Fredest9112) [reply app/src/main/java/com/learningkotlin/pokedex/model/PokemonModel.kt]: commented code removed

**Review by fredestReviewer — APPROVED** [review]

**Pull request merged**